src/box/box.cc

```diff
@@ -235,6 +237,45 @@ box_storage_init(void);
 static void
 box_broadcast_status(void);
 
+/** Broadcast this instance's ballot. */
+static void
+box_broadcast_ballot(void);
+
+/**
+ * A timer to broadcast the updated vclock. Doing this on each vclock update
+ * would be too expensive.
+ */
+static ev_timer box_broadcast_ballot_timer;
+
+/** Set a new interval for vclock updates in ballot. */
+static void
+box_update_broadcast_ballot_interval(double interval)
+{
+	static double ballot_broadcast_interval;
+	double timeout = ev_timer_remaining(loop(),
+					    &box_broadcast_ballot_timer);
+	timeout -= ballot_broadcast_interval;
+	timeout += interval;
+	ev_timer_stop(loop(), &box_broadcast_ballot_timer);
+	ev_timer_set(&box_broadcast_ballot_timer, timeout, interval);
+	ev_timer_start(loop(), &box_broadcast_ballot_timer);
+	ballot_broadcast_interval = interval;
+}
+
+/** A callback to broadcast updated vclock in ballot by timeout. */
+static void
+box_broadcast_ballot_on_timeout(ev_loop *loop, ev_timer *timer, int events)
+{
+	(void)loop;
+	(void)timer;
+	(void)events;
+	static struct vclock broadcast_vclock;
+	if (vclock_compare_ignore0(&broadcast_vclock, &replicaset.vclock) == 0)
+		return;
+	box_broadcast_ballot();
+	broadcast_vclock = replicaset.vclock;
```

Comment:
`vclock_copy()` (for now doesn't matter, but if in future it will get more complicated, better be ready).

Reply:
Oh, completely forgot about that. Sure.

```diff
diff --git a/src/box/box.cc b/src/box/box.cc
index 9a4ec2695..f18ae97f0 100644
--- a/src/box/box.cc
+++ b/src/box/box.cc
@@ -273,7 +273,7 @@ box_broadcast_ballot_on_timeout(ev_loop *loop, ev_timer *timer, int events)
        if (vclock_compare_ignore0(&broadcast_vclock, &replicaset.vclock) == 0)
                return;
        box_broadcast_ballot();
-       broadcast_vclock = replicaset.vclock;
+       vclock_copy(&broadcast_vclock, &replicaset.vclock);
 }
 
 /**
```